test/e2e/framework/test_context.go

```diff
@@ -309,6 +311,7 @@ func RegisterNodeFlags() {
 func RegisterStorageFlags() {
 	flag.StringVar(&TestContext.CSIImageVersion, "csiImageVersion", "", "overrides the default tag used for hostpathplugin/csi-attacher/csi-provisioner/driver-registrar images")
 	flag.StringVar(&TestContext.CSIImageRegistry, "csiImageRegistry", "quay.io/k8scsi", "overrides the default repository used for hostpathplugin/csi-attacher/csi-provisioner/driver-registrar images")
+	flag.StringVar(&TestContext.GCEPDCSIImage, "gcePDCSIImage", "", "overrides the default image used for GCE PD CSI images")
```

Comment:
This doesn't seem very extendable for when we want to support testing on any arbitrary CSI driver.  I didn't really like this when we added the CSIImage flags above either.  Can we just update the testing manifests whenever we want to test a new version?

Reply:
I see no reason why we can't have a flag for each driver that needs it. The manifest can contain the default prod version of the driver and the flag only needs to be specified if you're testing a snowflake/dev version of that particular driver.

Otherwise anyone that wants to test a new version of the driver using these tests has to manually (or write custom scripting) to update the manifests, rebuild the e2e binaries, and get the new manifests/binaries onto their e2e cluster. Contrast this with just pushing your image somewhere online, and using a flag to point to the new image.

Flags in this case would be configuration, AFAIK (I could be wrong) the manifests are built into the e2e test binary and can be considered "code." I don't want people (incl. me) to have to change "code" to test new configurations.

Let me know if you disagree or I'm getting some facts mixed up!

Reply:
I would prefer if we could come up with a way to be able to run the e2e tests against any CSI driver without having to compile the specific CSI driver manifests into the e2e binary.  Otherwise, there's no limit to how many driver-specific flags that will get added.

Reply:
Not sure if some of the e2e refactoring work that @pohly is doing will help with this

Reply:
Adding new flags is fine, *but* please don't do it in test_context.go. The consensus so far in the E2E refactoring discussion is that test-specific configuration should be defined in the code that defines the tests, i.e. not in the framework.

We also tried to come up with a naming scheme that makes it possible to group related options in a config file and avoids name collisions. "Just use some unique prefix with dots" seems to be the outcome, so in this case "storage.csi.gce.image" might be a better name.

Or "gce.storage.csi.image"? Depends a bit on how many more GCE-specific flags there will be in the end and whether its better to group them together.

The problem of how to handle extra data files is still open. The original goal was to use bindata for everything, but that won't work for files that aren't part of Kubernetes, and even for those I found quite a few places that still relied on the `-repo-root` parameter to find files - manifest files might have been one of those. I don't know how that currently works in practice when people don't run test/e2e out of the source repository.